src/lib.rs

```diff
@@ -58,8 +58,8 @@ pub trait Cache: Send + Sync {
         key: Self::Key,
         value: Self::Value,
         ttl: NonZeroU64,
-    ) -> anyhow::Result<Self::Value>;
-    async fn get<'a>(&'a self, key: &'a Self::Key) -> anyhow::Result<Self::Value>;
+    ) -> anyhow::Result<Option<Self::Value>>;
```

Comment:
I think it should be
```suggestion
    ) -> anyhow::Result<()>;
```

otherwise you're introducing non-compatible behaviour for memory and cloudflare implementations since ttlCache return Some or None depending on the presence of the value in cache already and cloudflare always returns Some.
It'll be simpler for now just return nothing